tests/integration/askPresentAndSquadJoin.test.ts

```diff
@@ -0,0 +1,219 @@
+          teamId: import.meta.env.VITE_SLACK_TEAM_ID,
+          birthday: dayjs.utc().add(2, "month").toDate(),
+        },
+      ]);
+
+      const eventId = "PI1_" + Date.now().toString();
+
+      await sendScheduleEvent(
+        scheduleEvent,
+        JSON.stringify({
+          eventId,
+          eventType: "askPresentAndSquadJoinFromTeam",
+          payload: {
+            team: import.meta.env.VITE_SLACK_TEAM_ID,
+            birthdayPerson: constants.birthdayPerson,
+            eventId,
+          },
+        }),
+      );
+
+      const message = await waitForDm(eventId);
+
+      expect(message.blocks?.length, "Message doesn't have 5 blocks").toBe(5);
+
+      expect(
+        message.blocks?.[1].text?.text,
+        "Block doesn't mention birthday person.",
+      ).toContain(`<@${constants.birthdayPerson}>`);
+
+      expect(
+        message.blocks?.[2].element?.action_id,
+        "Block doesn't contain input element",
+      ).toBe(additionalPresentIdeasInputActionId);
+
+      expect(
+        message.blocks?.[3].element?.action_id,
+        "Block doesn't contain squad join checkbox",
+      ).toBe(squadJoinCheckboxActionId);
+
+      expect(
+        message.blocks?.[4].elements?.[0].action_id,
+        "Block doesn't contain save button",
+      ).toBe(additionalPresentIdeasSaveButtonActionId);
+    },
+    timeout,
+  );
+
+  it(
+    "Should save additional present idea and squad join to db",
+    async () => {
+      await testDb.insert(users).values([
+        {
+          id: constants.userId,
+          teamId: constants.teamId,
+          birthday: new Date(),
+        },
+        {
+          id: constants.birthdayPerson,
+          teamId: constants.teamId,
+          birthday: new Date(),
+        },
+      ]);
+
+      const eventId = "PI2_" + Date.now().toString();
+
+      await sendSlackInteraction({
+        type: "block_actions",
+        user: {
+          id: constants.userId,
+          team_id: constants.teamId,
+        },
+        response_url: `${
+          import.meta.env.VITE_API_URL
+        }/slack/test-payload?testId=${eventId}`,
+        actions: [
+          {
+            action_id: additionalPresentIdeasSaveButtonActionId,
+            type: "button",
+            value: constants.birthdayPerson,
+          },
+        ],
+        state: {
+          values: {
+            additionalPresentIdeasInputBlockId: {
+              additionalPresentIdeasActionId: {
+                type: "plain_text_input",
+                value: constants.presentIdea,
+              },
+            },
+            squadJoinCheckboxBlockId: {
+              squadJoinCheckboxActionId: {
+                type: "checkboxes",
+                selected_options: [
+                  {
+                    value: "Sure",
+                  },
+                ],
+              },
+            },
+          },
+        },
+      });
+
+      const presentIdea = await vi.waitFor(
```

Comment:
There most be a similar waitfor in the other presentIdea related tests. Could you please reuse that here?

Reply:
Not sure how can i reuse that here, I'm doing the same.
<img width="2104" alt="image" src="https://github.com/theapexlab/birthday-slack-bot/assets/34692729/cbdc0c9f-e466-4db0-8a72-1df744b856f9">

Reply:
Or you mean outsorce it? its been use only 2 times, not sure worth it.